### PR TITLE
What is Software Process Improvement

### PR DESCRIPTION
**Pull request by bartlettroscoe**
# Description

EB Member: @rinkug 

Resolves #`<issue-id>`

`<Other minimal information about the PR.>`

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [ ] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* [ ] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [x] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [x] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [x] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [x] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [x] ***[Author]*** Make any final changes to the PR based on feedback.
* [x] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [x] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [ ] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [ ] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [x] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [x] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [x] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+2. **Establish goals and objectives:**  These goals should be specific, measurable, achievable, relevant, and time-bound.
+
+3. **Develop a plan:** This plan should include a timeline, resources, and responsibilities.
+
+4. **Implement the plan:** This involves implementing the changes (often couched as experiments) that have been identified and putting the collection of metrics in place to see if they are effective.
+
+5. **Monitor and evaluate the improvement effort:** This involves collecting metrics data on the effectiveness of the changes (i.e. the experiments) that have been implemented and making adjustments as needed based on feedback.
+
+By implementing software process improvement, organizations can:
```
Comment (rinkug) [inline Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:24]: Implementing Software Process Improvement (SPI) can bring several benefits to organizations such as:

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#### Contributed by  [Roscoe A. Bartlett](https://github.com/bartlett)
+#### Publication date: April 28, 2023
+
+<!--- deck start --->
+Software process improvement is the application of a systematic approach to the identification, evaluation, and improvement of software processes which is critical for computational science and engineering projects.
+<!--- deck end --->
+
+<!--- body start --->
+
+The goal of **Software Process Improvement** is a systematic methodology to improve the quality, efficiency, and effectiveness of software development and maintenance.
```
Comment (rinkug) [inline Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:11]: Software process improvement is the application of a systematic approach to the identification, evaluation, and improvement of software processes which is critical for computational science and engineering projects.
The overall  goal of software process improvement in scientific computing is to enhance the quality, efficiency, reliability and sustainability of software development in scientific research.

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+Improvement efforts can implemented in a variety of ways, but it typically involve the following steps:
```
Comment (rinkug) [inline Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:12]: There are various ways to implement improvement efforts, but they generally involve the following steps:

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+By implementing software process improvement, organizations can:
+
+* improve the quality of software products
+* reduce development costs
+* increase productivity
+* improve customer satisfaction
+* enhance organizational reputation, and
+* increase market share
+
```
Comment (rinkug) [inline Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:32]: - Improve the quality of software products
- Reduce development costs
- Increase productivity
- Improve customer satisfaction
- Enhance organizational reputation
- Increase market share

**Review by rinkug — COMMENTED** [review]
Made minor adjustments. Feel free to disagree and merge

**Comment by bartlettroscoe** [comment]
@rinkug, thanks for your suggestions.  They are all implemented in [`6ac58e0` (#1659)](https://github.com/betterscientificsoftware/bssw.io/pull/1659/commits/6ac58e02bd4dfc4b6697191196fe87cbd88cad64).

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md (hunk 2 above)
Comment (bartlettroscoe) [reply Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:14]: Implemented in 6ac58e02bd4dfc4b6697191196fe87cbd88cad64.

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md (hunk 3 above)
Comment (bartlettroscoe) [reply Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:15]: Implemented in 6ac58e02bd4dfc4b6697191196fe87cbd88cad64.

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md (hunk 4 above)
Comment (bartlettroscoe) [reply Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:35]: Implemented in 6ac58e02bd4dfc4b6697191196fe87cbd88cad64.

**Comment by bartlettroscoe** [comment]
FYI: This PR branch is not merging to 'preview' due to the GHA error:

```
/usr/bin/docker run --name c0442809c160f94e1484f82ffb70dec44add5_69bc53 --label 6c0442 --workdir /github/workspace --rm -e "INPUT_TYPE" -e "INPUT_HEAD_TO_MERGE" -e "INPUT_TARGET_BRANCH" -e "INPUT_GITHUB_TOKEN" -e "INPUT_LABEL_NAME" -e "INPUT_FROM_BRANCH" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/bssw.io/bssw.io":"/github/workspace" 6c0442:809c160f94e1484f82ffb70dec44add5

/usr/local/bundle/gems/octokit-4.14.0/lib/octokit/response/raise_error.rb:16:in `on_complete': POST https://api.github.com/repos/betterscientificsoftware/bssw.io/merges: 403 - Resource not accessible by integration // See: https://docs.github.com/rest/branches/branches#merge-a-branch (Octokit::Forbidden)
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/response.rb:[9](https://github.com/betterscientificsoftware/bssw.io/actions/runs/4852168960/jobs/8646822803?pr=1659#step:4:10):in `block in call'
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/response.rb:61:in `on_complete'
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/response.rb:8:in `call'
	from /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
	from /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/request/retry.rb:[13](https://github.com/betterscientificsoftware/bssw.io/actions/runs/4852168960/jobs/8646822803?pr=1659#step:4:14)0:in `call'
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/rack_builder.rb:[14](https://github.com/betterscientificsoftware/bssw.io/actions/runs/4852168960/jobs/8646822803?pr=1659#step:4:15)3:in `build_response'
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/connection.rb:387:in `run_request'
	from /usr/local/bundle/gems/faraday-0.17.0/lib/faraday/connection.rb:175:in `post'
	from /usr/local/bundle/gems/sawyer-0.8.2/lib/sawyer/agent.rb:94:in `call'
	from /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/connection.rb:[15](https://github.com/betterscientificsoftware/bssw.io/actions/runs/4852168960/jobs/8646822803?pr=1659#step:4:16)6:in `request'
	from /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/connection.rb:28:in `post'
	from /usr/local/bundle/gems/octokit-4.14.0/lib/octokit/client/commits.rb:[21](https://github.com/betterscientificsoftware/bssw.io/actions/runs/4852168960/jobs/8646822803?pr=1659#step:4:22)4:in `merge'
	from /action/lib/index.rb:21:in `<main>'
```

I don' think think that is actually a merge conflict.  That looks to be a permissions problem.  See:

* https://dev.to/callmekatootie/debug-resource-not-accessible-by-integration-error-when-working-with-githubs-graphql-endpoint-5bim#:~:text=This%20error%20basically%20talks%20about,not%20accessible%20by%20integration%20error.

(We have an internal email chain for this problem but I thought it would be good to archive this in this PR for future reference.)

**Comment by bartlettroscoe** [comment]
FYI: When I reverted the commit c5f4a05c395d3395fe99674cd92e120f91865374 with the revert commit 966e9ea5049ae16b06befe5b101a083d0f705a4b (as part of PR #1669) and then rebased the topic branch of this PR on top of ‘master’ and force pushed to this PR, the merge to ‘preview’ occurred just fine as shown [here](https://github.com/betterscientificsoftware/bssw.io/actions/runs/4855218595/jobs/8653526196?pr=1659).

**Comment by rinkug** [comment]
@bartlettroscoe : After seeing this on preview, something felt missing, so I added a concluding sentence. Pls accept/reject and then merge the PR

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe : After seeing this on preview, something felt missing, so I added a concluding sentence. Pls accept/reject and then merge the PR

@rinkug, I don't see your commit or suggested edits.

File: Articles/WhatIs/WhatIsSoftwareProcessImprovement.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+<!--- deck end --->
+
+<!--- body start --->
+
+The goal of **Software Process Improvement** is a systematic methodology to improve the quality, efficiency, and effectiveness of software development and maintenance.
+Improvement efforts can implemented in a variety of ways, but it typically involve the following steps:
+
+1. **Identify the need for improvement:**  This can be done by conducting a process assessment, which involves gathering data on the current state of the software development process.
+
+2. **Establish goals and objectives:**  These goals should be specific, measurable, achievable, relevant, and time-bound.
```
Comment (bartlettroscoe) [inline Articles/WhatIs/WhatIsSoftwareProcessImprovement.md:19]: This comment is just for demonstration.

**Review by bartlettroscoe — COMMENTED** [review]
This is my dummy review that will post my comments..

**Review by rinkug — APPROVED** [review]
LGTM

**Review by bartlettroscoe — COMMENTED** [review]
Hey @bartlettroscoe, this looks okay but please apply the changes in:

* https://github.com/bartlettroscoe/bssw.io/pull/11

**Comment by bartlettroscoe** [comment]
@rinkug, I will go ahead and dismiss your last approving review so that you can suggest your final changes.

**Comment by bartlettroscoe** [comment]
@rinkug, from my standpoint, this is ready to merge.

**Review by rinkug — APPROVED** [review]

**Pull request merged**